docs/source/middleware.rst

```diff
@@ -20,36 +56,10 @@ It is entirely possible that ``ContextMiddleware`` will be removed in the future
 It is also possible that authors will make some changes to the ``BaseHTTPMiddleware`` to fix this issue.
 I'd advise to only use ``RawContextMiddleware``.
 
-.. warning::
```

Comment:
I think this warning is still correct, so should not be removed. We only handle library errors. On 500 headers are not set on the response (sadly).

https://github.com/tomwojcik/starlette-context/blob/4550a190ac3e3b3c4dbdf9ba8733c7f78be80003/tests/test_context/test_context_on_unhandled_exception.py#L38-L40

Reply:
True, although at this point any Exception that arises would likely come from a user-implemented plugin not catching it.